pkgs/development/haskell-modules/lib.nix

```diff
@@ -299,12 +299,7 @@ rec {
     overrideCabal drv (_: { inherit src version; editedCabalFile = null; });
 
   # Get all of the build inputs of a haskell package, divided by category.
-  getBuildInputs = p:
-    (overrideCabal p (args: {
-      passthru = (args.passthru or {}) // {
-        _getBuildInputs = extractBuildInputs p.compiler args;
-      };
-    }))._getBuildInputs;
+  getBuildInputs = p: p.getBuildInputs;
```

Comment:
This change very slightly changes the results of this function, namely that `setupHaskellDepends` is not in `haskellBuildInputs` anymore. It is in the derivations `nativeBuildInputs` though, and therefore included in nix-shells.